src/Raven.Client/Json/Converters/StreamConverter.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+using Raven.Client.Extensions.Streams;
+using Sparrow.Json;
+
+namespace Raven.Client.Json.Converters
+{
+    internal sealed class StreamConverter : RavenJsonConverter
+    {
+        public static readonly StreamConverter Instance = new StreamConverter();
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            if (!(writer is BlittableJsonWriter blittableJsonWriter))
+            {
+                throw new SerializationException(
+                    $"Try to write {nameof(LazyStringValue)} property/field by {writer.GetType()} witch is unsuitable reader. Should use {nameof(BlittableJsonWriter)}");
+            }
+
+            if (!(value is Stream stream))
+            {
+                throw new SerializationException($"Try to write {value.GetType()} witch is not {nameof(Stream)} value with {nameof(StreamConverter)}");
+            }
+
+            if (false == stream.CanSeek)
+            {
+                throw new SerializationException("Try to serialize stream that can't seek");
+            }
+
+            stream.Seek(0, SeekOrigin.Begin);
+            var buffer = stream.ReadData();
+            stream.Seek(0, SeekOrigin.Begin);
+            var strBuffer = Convert.ToBase64String(buffer);
```

Comment:
Maybe we could consider reducing allocations here, by using LazyStringValue, that is being fed by Convert.ToBase64CharArray

Reply:
An issue created
[RavenDB-11852](http://issues.hibernatingrhinos.com/issue/RavenDB-11852)